### DISCUSSION TITLE
Would like to have the function of highlighting and deleting.

### DISCUSSION THREAD

**Discussion by AlbertWang257** [Eugeny/tabby#7314] [General]
It would be better if the highlighted parts were deleted, instead of deleting one word at a time.Sometimes the command is very long, and you just need to delete one of the parameters. In this case, you need to highlight the selected part and delete it instead of clicking the delete key all the time.

**Comment by Eugeny** [comment]
A terminal is just an interface for the programs (your shell) to talk through. Most programs running inside a terminal are not mouse-aware, and even fewer use it for text editing (e.g. vim's mouse mode). IIRC there was some ZSH plugin around that allowed mouse-controller cursor movement

**Reply by dizzy-egg** [reply]
I don't think the OP wants full mouse interactivity; it sounds like they just want to be able to highlight parts of whatever command they're currently typing so they can delete parts of it selectively.

It sounds very similar to what I'm looking for in my recent feature request:
https://github.com/Eugeny/tabby/issues/7389

I don't know the technical details of such functionality, but take a look at [SmarTTY](https://sysprogs.com/SmarTTY/). It allows manipulation of the command line string to achieve what the OP and myself are both looking for. An educated guess: it probably processes whatever you're typing locally and doesn't rely on the server receiving your text input as you type each character of the command. This also results in zero latency when typing commands, whereas I get pretty bad "input lag" when typing on Tabby or PuTTY.

Would be massively valuable to have such functionality in Tabby.

**Reply by AlbertWang257** [reply]
That's pretty much the idea, and I hope tabby gets better and better.
